src/core/createFrameResponse.ts

```diff
@@ -0,0 +1,58 @@
+import { NextResponse } from 'next/server';
+import { Button, FrameMetadata } from './sharedTypes';
+
+/**
+ * Creates a response with a new valid frame to be rendered.
+ *
+ * @param buttons: The buttons to use for the frame.
+ * @param image: The image to use for the frame.
+ * @param post_url: The URL to post the frame to.
+ * @param refresh_period: The refresh period for the image used.
+ * @returns An HTML response containing metadata for the frame.
+ */
+function createFrameResponse({
+  buttons,
+  image,
+  post_url,
+  refresh_period,
+}: FrameMetadata): NextResponse {
+  const imageHtml = image ? `<meta property="fc:frame:image" content="${image}" />` : '';
+  // Ensure only up to 4 buttons are processed
+  let buttonsHtml = '';
+  if (buttons) {
```

Comment:
Isn't the type already checking if the buttons exist? I think we can avoid this condition.

Reply:
I updated `FrameMetadata` to reflect the required/optional fields, so `buttons` is now optional, as it is in the Farcaster API.